src/Z3/RLock.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE BangPatterns #-}
+
+-- |
+-- Module    : Z3.RLock
+-- Copyright : (c) Iago Abal, 2013-2015
```

Comment:
If the code is mostly a copy paste from concurrent-extra, then the copyright notice should be:
```haskell
-- Copyright  : (c) 2010-2011 Bas van Dijk & Roel van Dijk
```